src/groupByKey.ts

```diff
@@ -2,6 +2,33 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+type Value = string | number;
+
+type Obj = {
+  [key: string]: Value;
+};
+
+export function groupByKey(items: Obj[], key: keyof Obj): GroupsMap<Obj> {
+  const result: GroupsMap<Obj> = {};
+  const moddedItems = items.map((item1) => {
+    const arrayForItems: Obj[] = [];
+
+    items.filter((item2) => {
+      if (item2[key] === item1[key]) {
+        arrayForItems.push(item2);
+
+        return true;
+      }
+
+      return false;
+    });
+
+    return arrayForItems;
+  });
+
+  items.forEach((item, index) => {
+    result[item[key]] = moddedItems[index];
+  });
```

Comment:
This loop is unnecessary because you can add items to the `result` object in the first loop. Also, the code `result[item[key]] = moddedItems[index];` overwrites the value of the `result[item[key]]` property, so the resulting object has the same keys and arrays for each key. This is not what the task requires. You need to add items to the `result[item[key]]` array, not overwrite it.